rootfs/ingest/python/qserv/metadata.py

```diff
@@ -251,25 +254,92 @@ def __init__(self, path: str, loadbalancers: List[str] = []):
             access to metadata. Defaults to [].
 
         """
-
+        self._database: str
+        self._json_db: Dict[str, Any] = {}
         self.fileformats: Dict[str, FileFormat] = {}
-        self.tables: List[TableSpec]
+        self._tableSpecs: List[TableSpec]
 
         # Get scheme configuration
         lbAlgo = LoadBalancerAlgorithm(loadbalancers)
         self.lb_url = LoadBalancedURL(path, lbAlgo)
 
         self.metadata_url = self.lb_url.direct_url
-        self.metadata = json_get(self.metadata_url, _METADATA_FILENAME)
+        self.metadata = json_load(self.metadata_url, _METADATA_FILENAME)
+        self._check_version()
 
         filename = self.metadata["database"]
-        self.json_db = json_get(self.metadata_url, filename)
-        self.database = self.json_db["database"]
-        self.family = "layout_{}_{}".format(self.json_db["num_stripes"], self.json_db["num_sub_stripes"])
+        self._json_db = json_load(self.metadata_url, filename)
```

Comment:
Why turning this member to be private? I have see it using directly by other module(s)? This violates principles of the OO programming.

See the relevant question I posted for method `database_register_and_config` of class `TransactionAction`

If you need to access the private member outside the current context then you may introduce the public accessor method like what you did for `_database`.